sbom/examples/product/create_product_sbom.py

```diff
@@ -142,36 +153,37 @@ def create_cdx():
             "timestamp": "2006-08-14T02:34:56Z",
             "tools": [{"name": "example tool", "version": "1.2.3"}],
         },
-        "components": [
-            product_component.copy() | {"bom-ref": cpe, "cpe": cpe} for cpe in product.cpes
-        ],
+        "components": [product_component.copy() | {"bom-ref": name_short}],
```

Comment:
We have to replicate it because TPA doesn't consider the metadata.component as a real component, see discussion in https://redhat-internal.slack.com/archives/C07MHR3MGLR/p1729693977566349.

Reply:
In addition to Martin's point about TPA, I think there's also the issue of being able to get the .metadata.component represented in nested components, if we don't want to always define it strictly as the root. By defining it also in the .components[] it also makes it possible for it to participate in the component tree. 

SPDX made it really simple with just relying on relationship types between nodes, including the top component also getting defined that way, but CycloneDX has some features like that that make it a bit awkward to do it the same way.